### ISSUE TITLE
8.2.4 removing Exemptions in OwnedOnly mode

### ISSUE THREAD

**Issue by robsissons-contino** [Azure/enterprise-azure-policy-as-code#364]
When building a policy plan with the latest Powershell module the plan shows that ALL existing Exemptions will be removed despite the PAC environment being configured for "ownedOnly" strategy.

EPAC Module version: 8.2.4
Powershell version: 7.3.6

Example:

**global-settings.json:**
```
{
    "$schema": "https://raw.githubusercontent.com/Azure/enterprise-azure-policy-as-code/main/Schemas/global-settings-schema.json",
    "pacOwnerId": "policy-as-code",
    "pacEnvironments": [
        {
            "pacSelector": "production",
            "cloud": "AzureCloud",
            "tenantId": "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
            "deploymentRootScope": "/providers/Microsoft.Management/managementGroups/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx",
            "desiredState": {
                "strategy": "ownedOnly"
            }
        }
    ]
}
```

**plan output snippets:**
```
===================================================================================================
Processing Policy Exemption files in folder 'C:\Repos\EPAC\azure-policy-as-code\production\Definitions\/policyExemptions/production'
===================================================================================================
Number of Policy Exemption files = 1
Processing file 'C:\Repos\EPAC\azure-policy-as-code\production\Definitions\policyExemptions\production\all-exemptions.csv'
Delete 'xxxxxxxx', '/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/rgxxxxxxxx'
Delete 'xxxxxxxx', '/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/rgxxxxxxxx'
Delete 'xxxxxxxx', '/subscriptions/xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx/resourceGroups/rgxxxxxxxx'
...
```

```
Policy Exemption counts:
    0 unchanged
    354 orphaned
    0 expired
    354 changes:
        new     = 0
        update  = 0
        replace = 0
        delete  = 354

```

**Comment by anwather** [comment]
Let me take a look at this tomorrow, just to confirm the exemptions aren't managed by EPAC?

**Comment by robsissons-contino** [comment]
correct - we currently have no exemptions managed by EPAC but do have a template CSV file in the directory which is why it shows "processing 1 file".

Tested with previous versions and no issues

**Comment by techlake** [comment]
Delete the directory as well

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by anwather** [comment]
I can replicate this by creating an assignment (portal), creating an exemption (portal) and then removing the assignment. The exemption shows up as orphaned - can you confirm that the assignments that the exemptions pertain to exist and are in scope? And are they managed by EPAC?

**Comment by anwather** [comment]
Yes please delete or rename the directory and let us know the outcome - I'm curious as to why those exemptions are showing up as orphaned, my understanding is an orphaned exemption is one where there is no assignment id available.

**Comment by robsissons-contino** [comment]
Done some further tests on this:

1. Removing or renaming the directory so no exemption files are processed removes the prompt in the build plan so no exemptions would be removed.

2. If I put the directory back and add a file with a single exemption for our EPAC instance to manage (still in ownedOnly) then the list of orphaned exemptions returns to the plan advising they would be deleted.

Also reviewed the current estate and can confirm that the exemptions being listed are correctly identified as "orphaned" as they are both expired and the policy which the exemption was created for no longer exists. Requested that the current policy team review and remove these however I still believe that when deploying in "ownedOnly" the existing exemptions should not be in scope for removal, regardless of their status.

**Note** - the build plan does not attempt to delete exemptions which are just expired and not orphaned (the definition which the exemption was created against still exists).

Hope this helps to triage.

**Comment by anwather** [comment]
My question would be if an exemption is orphaned would there be a valid case for leaving it deployed? The assignment is missing and it is expired so removing it makes sense..

Get Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by robsissons-contino** [comment]
Agree that these exemptions are no longer required and should be removed but they are not managed by EPAC and so when running in OwnedOnly mode they should not be in scope for EPAC to remove.

The team will manually remove these before we take full ownership of the environment with EPAC but still believe this is not the expected behaviour from EPAC
